x-pack/plugins/enterprise_search/server/routes/app_search/credentials.ts

```diff
@@ -42,9 +23,30 @@ export function registerCredentialsRoutes({
     },
     enterpriseSearchRequestHandler.createRequest({
       path: '/as/credentials/collection',
-      hasValidData: (body?: ICredentialsResponse) => {
-        return Array.isArray(body?.results) && typeof body?.meta?.page?.total_results === 'number';
-      },
     })
   );
+  router.get(
+    {
+      path: '/api/app_search/credentials/details',
+      validate: false,
+    },
+    enterpriseSearchRequestHandler.createRequest({
+      path: '/as/credentials/details',
+    })
+  );
```

Comment:
I know this is an ent-search thing, but I'm personally very confused on the difference between

http://localhost:3002/as/credentials/collection
and
http://localhost:3002/as/credentials/details

Do we really need `/details`? I know we likely don't need `lmAccount` or `apiUrl` data any longer... I guess we need `engines` data for a flyout (...although I wonder if we shouldn't just store that info separately in an EnginesLogic state?)

I guess these are better questions for the public API refactor. Sorry, this is a non-blocking comment I suppose, just me being confused out loud :)

Reply:
I also was surprised by this. I don't think we should need to call two API endpoints here.

Reply:
Hopefully we can clean it up at some point in our future API refactor! ✨